subworkflows/local/metagenomics_profiling.nf

```diff
@@ -0,0 +1,160 @@
+
+    if ( params.metagenomics_profiling_tool == 'malt' ) {
+
+        // Reset entire input meta for MALT to just database name,
+        // as we don't run run on a per-sample basis due to huge databases
+        // so all samples are in one run and so sample-specific metadata
+        // unnecessary. Set as database name to prevent `null` job ID and prefix.
+
+        if ( params.metagenomics_malt_group_size > 0 ) {
+            ch_input_for_malt_tmp =  reads
+                .map { meta, reads -> reads }
+                .collate( params.metagenomics_malt_group_size ) //collate into bins of defined lengths
+                .map{
+                    reads ->
+                    // add new meta with db-name as id
+                    [[id: file(params.metagenomics_profiling_database).getBaseName() ], reads]
+                }
+                .combine(database) //combine with database
+                .multiMap{
+                    // and split apart again
+                    meta, reads, database ->
+                        reads: [meta, reads]
+                        database: database
+                }
+            ch_input_for_malt = ch_input_for_malt_tmp.reads
+            database = ch_input_for_malt_tmp.database
+        }
+
+        else {
+            ch_input_for_malt =  reads
+                .map { meta, reads -> reads }
+                .collect()
+                .map{
+                    // make sure id is db_name for publishing purposes.
+                    reads ->
+                    [[id: file(params.metagenomics_profiling_database).getBaseName() ], reads]
+                }
+        }
+
+        // Run MALT
+        MALT_RUN ( ch_input_for_malt, database )
+
+        ch_maltrun_for_megan = MALT_RUN.out.rma6
+            .transpose()
+            .map {
+                meta, rma ->
+                    // re-extract meta from file names, use filename without rma to
+                    // ensure we keep paired-end information in downstream filenames
+                    // when no pair-merging
+                    def meta_new = meta.clone()
+                    meta_new['db_name'] = meta.id
+                    meta_new['id'] = rma.baseName
+                [ meta_new, rma ]
+            }
+
+        ch_versions                 = ch_versions.mix( MALT_RUN.out.versions.first() )
+        ch_raw_classifications      = ch_raw_classifications.mix( ch_maltrun_for_megan )
+        ch_multiqc_files            = ch_multiqc_files.mix( MALT_RUN.out.log )
+        ch_postprocessing_input     = ch_postprocessing_input.mix( ch_maltrun_for_megan )
+    }
+
+    else if ( params.metagenomics_profiling_tool == 'metaphlan' ) {
+
+        reads = reads.combine(database)
+        metaphlan_reads = reads.map{ meta, reads, database -> [meta, reads] }
+        metaphlan_db = reads.map{ meta, reads, database -> [database] }
+
+        METAPHLAN_METAPHLAN ( metaphlan_reads , metaphlan_db )
```

Comment:
I suspect you may want to do a multiMap here if this is necessary, I'm not following why the reads and database been combined here? Assuimg `database` is still a value channel, it's not necessary to combine with the reads I think